### PR TITLE
feat: Adding Favorites list

### PR DESCRIPTION
**Pull request by nickrandolph**
GitHub Issue (If applicable): #

## PR Type

What kind of change does this PR introduce?
- Feature

## What is the new behavior?

Added Deals & Favorites tabs on the DealsPage
Enabled Favorites toggle from product details page

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by nickrandolph** [comment]
@dr1rrb it's not a blocker for this PR but I need some help getting the Favorites list updated when the use toggles the isfavorite flag on the product. I suspect this just requires the use of messaging similar to todo.

File: samples/Commerce/Commerce.UI/Controls/AncestralBindingExtension.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+
+namespace Commerce.Controls
+{
+	/// <summary>
+	/// Provides a mean to bind to the data context of the ancestor of a specific type.
+	/// </summary>
+	public class AncestralBindingExtension : MarkupExtension
```
[the remark this answers is not in the document]
Comment (carldebilly) [reply samples/Commerce/Commerce.UI/Controls/AncestralBindingExtension.cs:11]: GitMerge
MERGED!!!

File: samples/Commerce/Commerce.UI/App.xaml [hunk 2]
```diff
@@ -33,8 +33,9 @@
 			<x:Double x:Key="LandscapeMaxPageWidth">500</x:Double>
 
 			<!-- To override the default FlyoutLightDismissOverlayBackground resource -->
-			<SolidColorBrush x:Key="FlyoutLightDismissOverlayBackground"
-							 Color="{StaticResource OverlayColor}" />
+			<!-- TODO: This override is breaking the flyouts -->
```
Comment (nickrandolph) [inline samples/Commerce/Commerce.UI/App.xaml:36]: @carldebilly I'm not sure what happened with the latest style changes but there's a few styles/resources that aren't resolving. Can you take a look at the TODO items in this PR to see what's missing?

File: samples/Commerce/Commerce.UI/Controls/AncestralBindingExtension.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (nickrandolph) [reply samples/Commerce/Commerce.UI/Controls/AncestralBindingExtension.cs:11]: All great, but there's no dev release for the Uno.Toolkit.UI. @jeromelaban  is there something preventing the CI from pushing new dev releases for the toolkit?

**Comment by nickrandolph** [comment]
@kazo0 I tried updating to latest Uno.Material.WinUI package (2.1.0-dev.5) and am seeing this error on app startup: System.ArgumentException: 'Missing resource: key=MaterialProgressRingStyle'
image
I didn't push this change since it prevents app from starting but if you can take a look to see what we need to do to resolve this error that'd be great

**Review by carldebilly — APPROVED** [review]

**Pull request merged**